ui/src/containers/CreateVolume.js

```diff
@@ -88,37 +101,44 @@ const CreateVolume = props => {
     { label: 'RawBlockDevice', value: RAW_BLOCK_DEVICE },
     { label: 'SparseLoopDevice', value: SPARSE_LOOP_DEVICE }
   ];
+
   const initialValues = {
     name: '',
     storageClass: storageClassesName[0],
     type: types[0].value,
-    size: '',
-    path: ''
+    path: '',
+    selectedUnit: sizeUnits[0].value,
+    sizeInput: ''
   };
-  const validationSchema = Yup.object().shape({
-    name: Yup.string()
+  const validationSchema = yup.object().shape({
+    name: yup
+      .string()
       .matches(
         /^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/,
         intl.messages.volume_name_error
       )
       .required(),
-    storageClass: Yup.string().required(),
-    type: Yup.string().required(),
-    path: Yup.string()
+    storageClass: yup.string().required(),
+    type: yup.string().required(),
+    path: yup
+      .string()
       .matches(/^\//, intl.messages.volume_path_error)
       .when('type', {
         is: RAW_BLOCK_DEVICE,
-        then: Yup.string().required()
+        then: yup.string().required()
       }),
-    size: Yup.string()
-      .matches(
-        /^([+-]?[0-9.]+)([eEinumkKMGTP]*[-+]?[0-9]*)$/,
-        intl.messages.volume_size_error
-      )
-      .when('type', {
-        is: SPARSE_LOOP_DEVICE,
-        then: Yup.string().required()
-      })
+    sizeInput: yup.number().when('type', {
+      is: SPARSE_LOOP_DEVICE,
+      then: yup
+        .number()
+        .integer(intl.messages.volume_size_error)
```

Comment:
This is way more readable than a regex :+1: 

That being said, it's buggy :disappointed: 

If you input a "big" number, say 2 GiB as bytes (which is pretty small actually), e.g. `2147483648`, you have the error `The size must be a postive integer.` which is pretty weird from a user point of view (it is a positive integer).

I guess we wouldn't have the problem with the good ol’ regex :stuck_out_tongue: